client/packages/android/app/src/main/java/org/openmsupply/client/ExtendedWebViewClient.java

```diff
@@ -0,0 +1,72 @@
+package org.openmsupply.client;
+
+import android.graphics.Bitmap;
+import android.webkit.WebView;
+
+import com.getcapacitor.Bridge;
+import com.getcapacitor.BridgeWebViewClient;
+import com.getcapacitor.JSExport;
+import com.getcapacitor.Logger;
+import com.getcapacitor.PluginHandle;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ExtendedWebViewClient extends BridgeWebViewClient {
+    Bridge bridge;
+
+    public ExtendedWebViewClient(Bridge bridge) {
+        super(bridge);
+
+        this.bridge = bridge;
+    }
+
+    // Have to manually inject Capacitor JS, this typically happens in
+    // WebViewLocalServer.handleProxyRequest
+    // but since it manually uses net.URL to fetch the content of request, this
+    // fails for self signed certificates
+    // and plugin definitions etc is not injected
+    @Override
+    public void onPageStarted(WebView webView,
+                              String url,
+                              Bitmap favicon) {
+        // TODO make sure this is only injected for pages in native bundle
+        // There is no way to get the full list of plugins from bridge, use 'debug' and
+        // see what plugins to add
+        List<PluginHandle> pluginList = Arrays.asList(bridge.getPlugin("NativeApi"), bridge.getPlugin("Keyboard"),
```

Comment:
There is a comment in README.md about this and link to capacitor discussion.

This is the only thing I don't like about this PR, and it took ages to figure out

Reply:
Yeah, I've got that `TODO` we can share current connected server with ExtendedWebViewClient and make sure to only inject there, although capacitor will still try to inject and if it's proper SSL cert, since allowNavigation is [*] (I think there is another way to do this with Capacitor, should be able to add a layer for allowed sites on top of allowNavigation permission), added this to be looked at in https://github.com/openmsupply/open-msupply/issues/743#issuecomment-1313398595